File: src/Python/101-200/118.YangHuiTriangle.py

class Solution:
    def generate_next(self, arr):
        n = len(arr)
        r = []
        r.append(1)
        for i in range(n - 1):
            r.append(arr[i] + arr[i + 1])
        r.append(1)
        return r

    def generate(self, numRows):
        """
        :type numRows: int
        :rtype: List[List[int]]
        """
        null = []
        one = [1]
        two = [1, 1]
        if numRows == 0:
            return null
        if numRows == 1:
            null.append(one)
            return null
        if numRows == 2:
            null.append(one)
            null.append(two)
            return null
        null.append(one)
        null.append(two)
        arr = two
        for i in range(numRows - 2):
            arr = self.generate_next(arr)
            null.append(arr)
        return null
        
if __name__ == '__main__':
    solution = Solution()
    print(solution.generate(0))
    print(solution.generate(1))
    print(solution.generate(2))
    print(solution.generate(5))
else:
    pass

File: src/CSharp/101-200/118.YangHuiTriangle.cs
using System.Collections.Generic;

namespace PeefyLeetCode.YangHuiTriangle
{
    public class Solution
    {
        public List<int> GenerateNext(List<int> arr)
        {
            var n = arr.Count;
            var r = new List<int>();
            r.Add(1);
            for (int i = 0; i < n - 1; i++)
            {
                r.Add(arr[i] + arr[i + 1]);
            }
            r.Add(1);
            return r;
        }
        public IList<IList<int>> Generate(int numRows)
        {
            var none = new List<IList<int>>();
            var one = new List<int>();
            one.Add(1);
            var two = new List<int>();
            two.Add(1);
            two.Add(1);
            if (numRows == 0)
                return none;
            if (numRows == 1)
            {
                none.Add(one);
                return none;
            }
            if (numRows == 2)
            {
                none.Add(one);
                none.Add(two);
                return none;
            }
            none.Add(one);
            none.Add(two);
            var arr = two;
            for (int i = 0; i < numRows - 2; i++)
            {
                arr = GenerateNext(arr);
                none.Add(arr);
            }
            return none;
        }
    }
}